aspnetapp/README.md

```diff
@@ -1,36 +1,95 @@
 
 ```console
-dotnet restore
-dotnet run
+cd aspnetapp
+dotnet run -c release
 ```
 
-Follow these steps to run this sample in a Linux or Windows container:
+After the application starts, visit `http://localhost:8000` in your web browser.
+
+Note: The `-c release` argument builds the application in release mode (the default is debug mode). See the [dotnet run reference](https://docs.microsoft.com/dotnet/core/tools/dotnet-run) for more information on commandline parameters.
+
+## Build and run the sample with Docker for Linux containers
+
+You can build and run the sample in Docker using Linux containers with the following commands. The instructions assume that you are in the root of the repository.
 
 ```console
+cd aspnetapp
 docker build -t aspnetapp .
-docker run -d -p 80:80 aspnetapp
+docker run -it --rm -p 8000:80 aspnetapp
 ```
 
-## View your web page running from your container
-* If you are using a linux container you can simply browse to http://localhost:80 to access your app in a web browser.
-* If you are using the Nano [Windows Container](https://docs.docker.com/docker-for-windows/) there is currently a bug that affects how [Windows 10 talks to Containers via "NAT"](https://github.com/Microsoft/Virtualization-Documentation/issues/181#issuecomment-252671828) (Network Address Translation). Today you must hit the IP of the container directly. You can get the IP address of your container with the following steps:
-  1. Run `docker ps` and copy the hash of your container ID
-  3. Run `docker inspect -f "{{ .NetworkSettings.Networks.nat.IPAddress }}" HASH` where `HASH` is replaced with your container ID.
-  4. Copy the container ip address and paste into your browser with port 80. (Ex: 172.16.240.197:80)
\ No newline at end of file
+After the application starts, visit `http://localhost:8000` in your web browser.
+
+Note: The `-p` argument maps port 8000 on you local machine to port 80 in the container (the form of the port mapping is `host:container`). See the [Docker run reference](https://docs.docker.com/engine/reference/commandline/run/) for more information on commandline paramaters.
+
+## Build and run the sample with Docker for Windows containers
+
+You can build and run the sample in Docker using Windows containers with the following commands. The instructions assume that you are in the root of the repository.
+
+```console
+cd aspnetapp
+docker build -t aspnetapp .
+docker run -d --rm aspnetapp
+```
+
+You must navigate to the container IP (as opposed to http://localhost) in your browser directly when using Windows containers. You can get the IP address of your container with the following steps:
+
+1. Run `docker ps` to get the hash for the sample container.
+1. Run `docker exec HASH ipconfig` where `HASH` is replaced with your container hash (for example, `bb9eb9446863`).
```

Comment:
@glennc and I talked about this. He suggested this approach. I came to think this is better. `docker exec` teaches you more generally useful concepts than Docker inspect, IMO. If you know ipconfig, then this approach should also feel more familiar. 

The `docker inspect` approach is useful and has the benefit of producing a more scoped value. I found the exec approach more compelling.